controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +181,59 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) createOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	} else if secret.UID == "" {
```

Comment:
This will not cover the case where the secret exists but does not contain an onboarding token

Reply:
from https://github.com/red-hat-storage/ocs-operator/pull/3075#discussion_r2011508516
that's an outside involvement and user needs to delete the secret to regenerate it.

Reply:
Why can't we be helpful here? What is the side effect?

Reply:
depends on how it was removed, if it was copied by admin and something else got it removed we'll generate new one which'll not match old one.

Reply:
will be checked and updated if empty.